drafts/str4d-stratum/draft1.rst

```diff
@@ -70,6 +70,49 @@ requests, to simplify their response parsing.
 .. [JSON-RPC-1.0] JSON-RPC.org. *JSON-RPC 1.0 Specifications*.
   URL: http://json-rpc.org/wiki/specification (visited on 2016-09-24).
 
+Error Objects
+~~~~~~~~~~~~~
+
+The [JSON-RPC-1.0]_ specification allows for error objects in responses, but
+does not specify their format. The original Stratum protocol uses the following
+format for error responses [Slushpool-Stratum]_::
+
+    {"id": ##, "result": null, "error": [ERROR_CODE, "ERROR_MESSAGE", TRACEBACK]}\n
+
+For compatibility, this format is retained. We therefore define an error object
+as an array::
+
+    [ERROR_CODE, "ERROR_MESSAGE", TRACEBACK]
+
+``ERROR_CODE`` (int)
+  Indicates the type of error that occurred.
+
+  The error codes are to be interpreted as described in [JSON-RPC-2.0]_.
+  The following application error codes are defined:
+
+  - 20 - Other/Unknown
+  - 21 - Job not found (=stale)
+  - 22 - Duplicate share
+  - 23 - Low difficulty share
+  - 24 - Unauthorized worker
+  - 25 - Not subscribed
+
+``ERROR_MESSAGE`` (str)
+  A human-readable error message. The message SHOULD be limited to a concise
+  single sentence. Miners SHOULD display the message to the user.
```

Comment:
If the user interface language is not English, then an English-language message is going to look out of place (there's no provision to negotiate a language, and adding such provision would be overcomplication). I suggest saying that the miner software SHOULD display a human-readable message to the user, and that this message can be derived from either the ERROR_CODE or ERROR_MESSAGE fields, or both.